### DISCUSSION TITLE
Incorrect inference results

### DISCUSSION THREAD

**Discussion by as-stevens** [mindee/doctr#880] [Q&A]
Hi All,

I am trying to read the images as text using the below code:

```
image_path = 'test_sample.jpg'
img = read_img_as_tensor(image_path).unsqueeze(0)
device = torch.device("cuda:1" if torch.cuda.is_available() else "cpu")
predictor = ocr_predictor('db_resnet50', 'crnn_vgg16_bn', pretrained=True).cuda(device)
img = img.cuda(device)
result = predictor(img)

page = result.pages[0]
        #print(page)
height, width = page.dimensions
page_media = (0, 0, width, height)
for b in page.blocks:
    for line in b.lines:
    print(line)
```

The inference results are not good:

- The words are getting concatenated - for e.g. Membership interest inferred as Membershiplnterest
- Wrong characters identified- for e.g. Membership interest inferred as Membershiplnterest -note i from interest changed to l
- Missing text - some block of text is not read at all

Please let me if I need to do any pre-processing on the image or any other steps so that my results are more reliable.

**Comment by felixdittrich92** [comment]
Hi @as-stevens :wave: ,

This is a known issue so i think it would be great if you can wrote this also into #330  and add one or two examples if possible ? :hugs: 

Do you have two GPUs on your device ? Otherwise you can remove all without `ocr_predictor(...).cuda()` and have you tried to read your image with `img = DocumentFile.from_images([image_path])` ? :smile:
